src/main/java/teammates/ui/controller/AdminActivityLogPageAction.java

```diff
@@ -173,116 +166,36 @@ private void generateStatusMessage(AdminActivityLogPageData data, List<ActivityL
     }
     
     private List<ActivityLogEntry> searchLogsWithExactTimePeriod(AdminActivityLogPageData data) {
-        totalLogsSearched = 0;
-        LogQuery query = buildQuery(data);
-        List<ActivityLogEntry> appLogs = searchLogsByQuery(query, data);
+        LogHelper logHelper = new LogHelper();
+        logHelper.setQuery(data.getVersions(), data.getFromDate(), data.getToDate(), null);
+        
+        //LogQuery query = buildQuery(data);
+        List<ActivityLogEntry> appLogs = searchLogsByQuery(logHelper, data);
         return appLogs;
     }
 
-    private List<ActivityLogEntry> searchLogsByQuery(LogQuery query, AdminActivityLogPageData data) {
+    private List<ActivityLogEntry> searchLogsByQuery(LogHelper logHelper, AdminActivityLogPageData data) {
         List<ActivityLogEntry> appLogs = new LinkedList<ActivityLogEntry>();
-        //fetch request log
-        Iterable<RequestLogs> records = LogServiceFactory.getLogService().fetch(query);
-        for (RequestLogs record : records) {
-            //fetch application log
-            List<AppLogLine> appLogLines = record.getAppLogLines();
-            for (AppLogLine appLog : appLogLines) {
-                totalLogsSearched++;
-                String logMsg = appLog.getLogMessage();
-                if (logMsg.contains("TEAMMATESLOG") && !logMsg.contains("adminActivityLogPage")) {
-                    ActivityLogEntry activityLogEntry = new ActivityLogEntry(appLog);                   
-                    activityLogEntry = data.filterLogs(activityLogEntry);
-                    if (activityLogEntry.toShow() && ((!activityLogEntry.isTestingData()) || data.getIfShowTestData())) {
-                        if (isFirstRow ) {
-                            activityLogEntry.setFirstRow();
-                            isFirstRow = false;
-                        }
-                        appLogs.add(activityLogEntry);
+        List<AppLogLine> appLogLines = logHelper.fetchLogs();
+        totalLogsSearched = appLogLines.size();
+        
+        for (AppLogLine appLog : appLogLines) {
+            String logMsg = appLog.getLogMessage();
+            if (logMsg.contains("TEAMMATESLOG") && !logMsg.contains("adminActivityLogPage")) {
+                ActivityLogEntry activityLogEntry = new ActivityLogEntry(appLog);                   
+                activityLogEntry = data.filterLogs(activityLogEntry);
+                if (activityLogEntry.toShow() && ((!activityLogEntry.isTestingData()) || data.getIfShowTestData())) {
+                    if (isFirstRow ) {
+                        activityLogEntry.setFirstRow();
+                        isFirstRow = false;
                     }
+                    appLogs.add(activityLogEntry);
                 }
-            }    
+            }
```

Comment:
you can get rid of most of this nesting by reversing the condition and using an early termination. e.g.
```java
if(!condition){
    continue;
}
```